### ISSUE TITLE
Strange result with `L.Linear`, `to_gpu`, and `zerograds`.

### ISSUE THREAD

**Issue by toslunar** [chainer/chainer#6119]
* Conditions
```
>>> chainer.print_runtime_info()
Platform: Linux-4.4.0-103-generic-x86_64-with-debian-stretch-sid
Chainer: 6.0.0b2
NumPy: 1.17.0.dev0+8f547f2
CuPy:
  CuPy Version          : 6.0.0b2
  CUDA Root             : /usr/local/cuda
  CUDA Build Version    : 9000
  CUDA Driver Version   : 9000
  CUDA Runtime Version  : 9000
  cuDNN Build Version   : 7104
  cuDNN Version         : 7104
  NCCL Build Version    : 2005
iDeep: 2.0.0.post3
```

* Code to reproduce
```
>>> l = chainer.links.Linear(3, 4)
>>> l.zerograds()
>>> l.W.grad
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]], dtype=float32)
>>> l = chainer.links.Linear(3, 4).to_gpu(0)
>>> l.zerograds()
>>> l.W.grad
array([[nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan]], dtype=float32)
```

**Comment by fiarabbit** [comment]
This can be observed in v6.0.0b1, but is not observed in 5.1.0.

```
>>> chainer.print_runtime_info()
Platform: Linux-4.4.0-135-generic-x86_64-with-Ubuntu-16.04-xenial
Chainer: 5.2.0
NumPy: 1.15.4
CuPy:
  CuPy Version          : 6.0.0b1
  CUDA Root             : /home/hashimoto/.local/cuda/cuda-9.2
  CUDA Build Version    : 9020
  CUDA Driver Version   : 10000
  CUDA Runtime Version  : 9020
  cuDNN Build Version   : 7401
  cuDNN Version         : 7401
  NCCL Build Version    : 2307
iDeep: Not Available
>>> l = chainer.links.Linear(3, 4)
>>> l.zerograds()
>>> l.W.grad
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]], dtype=float32)
>>> l = chainer.links.Linear(3, 4).to_gpu(0)
>>> l.zerograds()
>>> l.W.grad
array([[nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan]], dtype=float32)
```

```
>>> chainer.print_runtime_info()
Platform: Linux-4.4.0-135-generic-x86_64-with-debian-stretch-sid
Chainer: 5.1.0
NumPy: 1.16.0
CuPy:
  CuPy Version          : 5.1.0
  CUDA Root             : /home/hashimoto/.local/cuda/cuda-9.2
  CUDA Build Version    : 9020
  CUDA Driver Version   : 10000
  CUDA Runtime Version  : 9020
  cuDNN Build Version   : 7301
  cuDNN Version         : 7301
  NCCL Build Version    : 2307
iDeep: Not Available
>>> l = chainer.links.Linear(3, 4)
>>> l.zerograds()
>>> l.W.grad
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]], dtype=float32)
>>> l = chainer.links.Linear(3, 4).to_gpu(0)
>>> l.zerograds()
>>> l.W.grad
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]], dtype=float32)
```

**Comment by fiarabbit** [comment]
`Parameter._grad` should be reference to `Parameter._grad_var.array` when `W._grad_var` is not `None`, but it is not the case when `to_gpu()`.

```
>>> l = chainer.links.Linear(3, 4)
>>> W = l.W
>>> W._grad
array([[nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan]], dtype=float32)
>>> W._grad_var is None
True
>>> W.to_gpu()
>>> W._grad
array([[nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan],
       [nan, nan, nan]], dtype=float32)
>>> W._grad_var
variable([[nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan],
          [nan, nan, nan]])
>>> W._grad is W._grad_var.array
False
```

**Comment by niboshi** [comment]
Fixed via #6170.
@fiarabbit Thank you for investigation!
